xbmc/addons/gui/skin/SkinTimerManager.cpp

```diff
@@ -23,39 +23,39 @@ using namespace std::chrono_literals;
 
 void CSkinTimerManager::LoadTimers(const std::string& path)
 {
-  CXBMCTinyXML doc;
+  CXBMCTinyXML2 doc;
   if (!doc.LoadFile(path))
   {
-    CLog::LogF(LOGWARNING, "Could not load timers file {}: {} (row: {}, col: {})", path,
-               doc.ErrorDesc(), doc.ErrorRow(), doc.ErrorCol());
+    CLog::LogF(LOGWARNING, "Could not load timers file {}: {} (Line: {})", path, doc.ErrorStr(),
+               doc.ErrorLineNum());
     return;
   }
 
-  TiXmlElement* root = doc.RootElement();
+  auto* root = doc.RootElement();
   if (!root || !StringUtils::EqualsNoCase(root->Value(), "timers"))
   {
     CLog::LogF(LOGERROR, "Error loading timers file {}: Root element <timers> required.", path);
     return;
   }
 
-  const TiXmlElement* timerNode = root->FirstChildElement("timer");
+  const auto* timerNode = root->FirstChildElement("timer");
   while (timerNode)
   {
     LoadTimerInternal(timerNode);
     timerNode = timerNode->NextSiblingElement("timer");
   }
 }
 
-void CSkinTimerManager::LoadTimerInternal(const TiXmlElement* node)
+void CSkinTimerManager::LoadTimerInternal(const tinyxml2::XMLNode* node)
 {
-  if ((!node->FirstChild("name") || !node->FirstChild("name")->FirstChild() ||
-       node->FirstChild("name")->FirstChild()->ValueStr().empty()))
+  if (!node->FirstChildElement("name") || !node->FirstChildElement("name")->FirstChild() ||
+       (strcmp(node->FirstChildElement("name")->FirstChild()->Value(), "\0") != 0))
```

Comment:
```suggestion
       (strcmp(node->FirstChildElement("name")->FirstChild()->Value(), "\0") == 0))
```

Reply:
I think this would be better if wrapped in some utils function `ElementNotEmpty(xx)` or similar. This is hard to read and really prone to errors.

Reply:
The problem is it isnt always just an empty test. theres many comparisons against values that will use the same strcmp. I guess you could use StringUtils::EqualsNoCase or something, but im not personally really sure why strcmp isnt adequate (apart from my copy pasta failures)

If you want easy, deal with std::string creation, but im not sure about the performance hit. Maybe @neo1973 can comment on that sort of thing

Reply:
Hmm, so many choices... For a simple empty check these would be the fastest and the room for error is small:
```c++
*node->FirstChildElement("name")->FirstChild()->Value() == '\0';
!*node->FirstChildElement("name")->FirstChild()->Value();
```
Readability could be improved by using `std::string_view` at the cost of a small overhead depending on the compiler (clang generates the exact same code for all 5 variants, GCC doesn't):
```c++
std::string_view(node->FirstChildElement("name")->FirstChild()->Value()).empty();
std::string_view(node->FirstChildElement("name")->FirstChild()->Value()) == "";
node->FirstChildElement("name")->FirstChild()->Value() == ""sv;  
```
`std::string_view` can also be used for comparisons in general but it will be slightly slower than `std::strcmp`.

I think I would use `*node->FirstChildElement("name")->FirstChild()->Value() == '\0'` for the empty check and `std::strcmp` for string comparison.

[Here](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGISQAcpK4AMngMmAByPgBGmMT%2BAKykAA6oCoRODB7evv5BaRmOAmER0SxxCZLJdpgOWUIETMQEOT5%2BgbaY9sUMjc0EpVGx8Um2TS1teZ0KE4PhwxWj1QCUtqhexMjsHOYAzOHI3lgA1CZ7brPE4cAA%2BgBueJgA7ufYJhoAgvuHx5hnF2QVxubw%2B3y%2BXgyRhOzDYChSTG2J1m6BAIFohHiYgUaOBRgeT2etwxBCxtAU5ysXzBaAYsxOyAQzQAVCcWEwANaYRrECArSlgsExVCeE5ULh8s4AdipnxO8oZAnpjJZyIBABE2ZzuQRefy9rKFSdiJgCJsGGrzuqrWczGZzGYFPcBdSpdbqV9haKqGZJSYZWCjbTlUziKyFBqtVyeXyXXKFSazcQLWAwMyKQbBW7BZ6RbQxXs/QGvkGlQQGaHw5H2dHdbHMyWE6bzciCKiQHi7o8XhAFCsAHSYFgpAgAT3rsv97vBny9%2BaokiLhoVwfLKrDlr2mprOr1caNiZb6Y1NrAu0Sbg0Z7jU5zs7zYsSS8DK7LFdVEZtO5j%2BuX8sPyatu2nYEj2fYnlutr2naN7Zh697egAbM%2Bjbyqu74bp%2BkHfnWv4vv%2BzaASiuK6sgw69qQUEABLdLQqAnAA6iQtDoA6KwQZqGiwdOQoPlQUoofGaFvuuVZftqP77k2SYWlh1qQQ6NG0HRjHMaxdpOtxd5zmKASCaWdJrpWm7bhJuFSQRMlASR1z4t2zy9uxVo2optH0UxxAsQ63EcGstCcIkvB%2BBwWikKgnBuNY1jIhsWz/PsPCkAQmi%2BWsCCYEwWAJHypAciAex7P2eySIhUoBFwZgVQEkgAJx1fonCSEFKVhZwvA4hoSUpWscCwEgaDDnQ8TkJQA0pENCRHIYwBcIhGidVgjzbAAaoSADyKSMJwiU0LQpLEDiEAxC1MThM0o7bbwp3MMQo5rTE2h1Ml3C8ANbCCGtDC0BdIW8FgMReMAbhiOSl2kFg7JGOIv3g3gJr1PcmA4jDmCqHUXiki14Skv5MMYjExDnR4WAtbqeAsJdaxUAYwAKKtLwbVtL0yIIIhiOwUgs/IShqC1uhcPo00oFFlj6HgMQ4pAayoCOWTIwAtCiVqmJY1hcFKJzy2tey8KgiPENcWCSzltT1M4ECuFMfhcDVwQMOgQzlJUIDJIUmQCFbEi227vSOyMCQ1N0T0NHMns210PQhwMftLAH4wDGHtuzNHCxO6MiRrAosXbBIfkBc1MPhRwJyqAEiHy4hkgMgY0Kzf2Gj1ycEC4IQJC2nsXArLwz1aCsaUZVllBrHlHf9iVZWJAENUBFKtVcAEQS401pAU1w82kMFoVF%2B1ICdT3vmkL1iAoKgg30GQFAQGNE0gDNexmODmBLZg9PPIzwU7XQ%2B2HcdMPXedYN/63Xuo9BwYM3qMAIJ9b6LV/qA2BspZGiUIbTWhqFfA8NHCI2RqFVG6NMbM2xt0Fq%2BNCa3WJjsUKZMKYvSpjTOm61Nof14PwVmohxCc1YdzFQ6gYa6AfjXYwIsbD42NtLWWSpOCKzbMrYR6tNba11vrQ2SN4CZyDmbPwFt7Zhztg7VO/s9A%2ByyLo4xJQDGxz0KbXo/RJieHaFYjRNi5gx2dgLZOdjcjW3ji0VxoxO7rE2DnAJuNAobxakXEuZcK5V2AMgZAJwuCFTME3FuRBiDt07t3bqw98qJH7PPRIexEg1SlHVRCk8pDSCXgXLebVbC7y6r9PuDUOBmF4Kvdem9db1P3i0/WGRnCSCAA%3D%3D%3D) the comparison of the different methods for the interested.